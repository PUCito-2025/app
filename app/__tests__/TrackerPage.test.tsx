import { render } from "@testing-library/react";
import { expect, test, vi } from "vitest";

import StudyTracker from "@/app/tracker/page";

// Mock the useUser hook from Clerk
vi.mock("@clerk/nextjs", () => ({
  useUser: () => ({
    user: {
      id: "test-user-id",
      fullName: "Test User",
    },
    isLoaded: true,
  }),
}));

// Mock fetch for API calls
global.fetch = vi.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve({
        courses: [],
        studyPlans: [],
      }),
  }),
) as unknown as typeof fetch;

test("TrackerPage matches snapshot", () => {
  const { container } = render(<StudyTracker />);
  expect(container).toMatchSnapshot();
});
