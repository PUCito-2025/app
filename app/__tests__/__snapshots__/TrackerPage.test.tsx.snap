// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TrackerPage matches snapshot 1`] = `
<div>
  <div
    class="p-6 space-y-8"
  >
    <div
      class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-4 gap-4"
    >
      <div
        class="rounded-2xl shadow-md bg-white p-4 bg-white shadow rounded-2xl p-4"
      >
        <div
          class="pt-2"
        >
          <h2
            class="text-xl font-bold mb-2"
          >
            Matemática
          </h2>
          <p>
            Horas recomendadas: 
            6
          </p>
          <p>
            Horas estudiadas: 
            2
          </p>
          <p>
            Horas restantes: 
            4
          </p>
          <div
            class="mt-2 flex space-x-2"
          >
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] disabled:opacity-50 md:text-sm aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
              data-slot="input"
              min="0"
              placeholder="Horas"
              type="number"
              value=""
            />
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md transition-all disabled:pointer-events-none disabled:opacity-50[&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0[&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px]aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
              data-slot="button"
            >
              Registrar
            </button>
          </div>
          <p
            class="mt-2 text-red-600 font-semibold"
          >
            ¡Alerta! Hay prueba la próxima semana.
          </p>
        </div>
      </div>
      <div
        class="rounded-2xl shadow-md bg-white p-4 bg-white shadow rounded-2xl p-4"
      >
        <div
          class="pt-2"
        >
          <h2
            class="text-xl font-bold mb-2"
          >
            Historia
          </h2>
          <p>
            Horas recomendadas: 
            4
          </p>
          <p>
            Horas estudiadas: 
            1
          </p>
          <p>
            Horas restantes: 
            3
          </p>
          <div
            class="mt-2 flex space-x-2"
          >
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] disabled:opacity-50 md:text-sm aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
              data-slot="input"
              min="0"
              placeholder="Horas"
              type="number"
              value=""
            />
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md transition-all disabled:pointer-events-none disabled:opacity-50[&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0[&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px]aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
              data-slot="button"
            >
              Registrar
            </button>
          </div>
        </div>
      </div>
      <div
        class="rounded-2xl shadow-md bg-white p-4 bg-white shadow rounded-2xl p-4"
      >
        <div
          class="pt-2"
        >
          <h2
            class="text-xl font-bold mb-2"
          >
            Inglés
          </h2>
          <p>
            Horas recomendadas: 
            3
          </p>
          <p>
            Horas estudiadas: 
            2
          </p>
          <p>
            Horas restantes: 
            1
          </p>
          <div
            class="mt-2 flex space-x-2"
          >
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] disabled:opacity-50 md:text-sm aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
              data-slot="input"
              min="0"
              placeholder="Horas"
              type="number"
              value=""
            />
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md transition-all disabled:pointer-events-none disabled:opacity-50[&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0[&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px]aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
              data-slot="button"
            >
              Registrar
            </button>
          </div>
        </div>
      </div>
      <div
        class="rounded-2xl shadow-md bg-white p-4 bg-white shadow rounded-2xl p-4"
      >
        <div
          class="pt-2"
        >
          <h2
            class="text-xl font-bold mb-2"
          >
            Software
          </h2>
          <p>
            Horas recomendadas: 
            5
          </p>
          <p>
            Horas estudiadas: 
            4
          </p>
          <p>
            Horas restantes: 
            1
          </p>
          <div
            class="mt-2 flex space-x-2"
          >
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] disabled:opacity-50 md:text-sm aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
              data-slot="input"
              min="0"
              placeholder="Horas"
              type="number"
              value=""
            />
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md transition-all disabled:pointer-events-none disabled:opacity-50[&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0[&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px]aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
              data-slot="button"
            >
              Registrar
            </button>
          </div>
          <p
            class="mt-2 text-red-600 font-semibold"
          >
            ¡Alerta! Hay prueba la próxima semana.
          </p>
        </div>
      </div>
    </div>
    <div
      class="mt-10"
    >
      <h2
        class="text-2xl font-bold mb-4"
      >
        Calendario de estudio
      </h2>
      <div
        class="grid grid-cols-7 gap-2 text-center"
      >
        <div
          class="bg-blue-100 rounded-xl p-2"
        >
          Día 
          1
          <div
            class="text-sm mt-1"
          >
            Horas: 3
          </div>
        </div>
        <div
          class="bg-blue-100 rounded-xl p-2"
        >
          Día 
          2
          <div
            class="text-sm mt-1"
          >
            Horas: 3
          </div>
        </div>
        <div
          class="bg-blue-100 rounded-xl p-2"
        >
          Día 
          3
          <div
            class="text-sm mt-1"
          >
            Horas: 3
          </div>
        </div>
        <div
          class="bg-blue-100 rounded-xl p-2"
        >
          Día 
          4
          <div
            class="text-sm mt-1"
          >
            Horas: 3
          </div>
        </div>
        <div
          class="bg-blue-100 rounded-xl p-2"
        >
          Día 
          5
          <div
            class="text-sm mt-1"
          >
            Horas: 3
          </div>
        </div>
        <div
          class="bg-blue-100 rounded-xl p-2"
        >
          Día 
          6
          <div
            class="text-sm mt-1"
          >
            Horas: 3
          </div>
        </div>
        <div
          class="bg-blue-100 rounded-xl p-2"
        >
          Día 
          7
          <div
            class="text-sm mt-1"
          >
            Horas: 3
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
